### PR TITLE
Fix incorrect URLs for next/previous URLs 

### PR DESCRIPTION
**Pull request by AbegaM**
# Purpose of PR

* This pull request addresses an issue with the `/tables/invoices/rows` API endpoint. When a request is sent to this endpoint, the response includes the data in the `invoices` column, as well as `next` and `prev` keys that allow the user to navigate between pages of results. However, as seen in the example response below, the URL parameters in the `next` field are incorrect. Specifically, the `page` parameter should be updated to `_page`, and if a user passes other parameters such as `_ordering`, `_filters`, `_limit`, `_schema`, `_extend`, and `_search`, these parameters are not displayed in the `next` and `prev` fields.

Example response:
```json
  {
    "data": [
        {
            "InvoiceId": 1,
            "CustomerId": 2,
            "InvoiceDate": "2009-01-01 00:00:00",
            "BillingAddress": "Theodor-Heuss-Straße 34",
            "BillingCity": "Stuttgart",
            "BillingState": null,
            "BillingCountry": "Germany",
            "BillingPostalCode": "70174",
            "Total": 1.98
        }
    ],
    "total": 412,
    "next": "/tables/invoices?page=2",
    "previous": null
}
```
# Modification

* To address this issue, I have modified the code in the `core/src/controllers/rows.js` file. Specifically, I have ensured that all query parameters that are added by the user are included in the `next` and `prev` fields of the API response. This ensures that users can properly navigate between pages of results, even when additional query parameters are present.

* To test this PR you can use the URL below 
```js
 localhost:8000/api/tables/invoices/rows?_filters=BillingCountry:Brazil&_search=SP&_ordering=-Total&_schema=customerId&_extend=CustomerId
```

### CODE REVIEW

**Comment by AbegaM** [comment]
@thevahidal and @IanMayo Please review this PR, i have fixed the errors in the `next`/`prev` fields

**Review by IanMayo — CHANGES_REQUESTED** [review]
Code reviewed, and API tested.

But, @AbegaM - how about adding a test in `rows.test.js`?

The test will verify that query params are passed to the `prev` and `next` outputs.

**Review by IanMayo — COMMENTED** [review]
Note: I get this error when I run the test locally:
```
    expect(received).toEqual(expected) // deep equality

    Expected: "/api/tables/users/rows?_limit=8&_schema=firstName%2ClastName&_ordering=-firstName&_search=a&_page=3"
    Received: "/tables/users/rows?_search=a&_ordering=-firstName&_schema=firstName,lastName&_limit=8&_page=3"

      38 |     expect(res.body.data[0]).toHaveProperty('lastName');
      39 |
    > 40 |     expect(res.body.next).toEqual(
         |                           ^
      41 |       `${baseUrl}?${new URLSearchParams({
      42 |         ...params,
      43 |         _page: params._page + 1,

      at Object.toEqual (src/controllers/rows.test.js:40:27)
```

**Comment by AbegaM** [comment]
@IanMayo can you please retest it now?

**Review by IanMayo — APPROVED** [review]
Tests pass :-D

But, a conflict to resolve, please @AbegaM

**Review by IanMayo — APPROVED** [review]

File: core/src/controllers/rows.test.js [hunk 1]
```diff
@@ -3,6 +3,14 @@ const supertest = require('supertest');
...
 const requestWithSupertest = supertest(app);
 
+function queryString(params) {
+  const queryString = Object.keys(params)
+    .map((key) => `${key}=${params[key]}`)
+    .join('&');
+
+  return queryString;
+}
+
```
Comment (thevahidal) [inline core/src/controllers/rows.test.js:13]: You can use `URLSearchParams` for this

**Review by thevahidal — CHANGES_REQUESTED** [review]

File: core/src/controllers/rows.test.js (hunk 1 above)
Comment (AbegaM) [reply core/src/controllers/rows.test.js:13]: When I pushed the first commit, the code was using URLSearchParams, which encodes the params object and adds encoding values like %2C and others when it encounters , and space. This encoding made it difficult to check if the URL I defined in the code is equal to the response res.body.next/res.body.previous. To address this issue, I added a queryString function that does not encode the URL, allowing for easier comparison and checking.

Comment (thevahidal) [reply core/src/controllers/rows.test.js:13]: Aha, I see, then it's good to go man!
Thanks for another awesome PR! You're killing it!

**Review by thevahidal — APPROVED** [review]
Thanks to @IanMayo, who reviewed it, this PR can move along!

File: core/src/controllers/rows.test.js (hunk 1 above)
Comment (AbegaM) [reply core/src/controllers/rows.test.js:13]: Thank you @thevahidal, I appreciate it

**Pull request merged**